features/dd-sdk-android-trace/src/test/kotlin/com/datadog/trace/api/IdGenerationStrategyTest.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.trace.api
+
+import com.datadog.android.utils.forge.Configurator
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.mock
+import org.mockito.quality.Strictness
+import java.security.SecureRandom
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class IdGenerationStrategyTest {
+
+    // Please note that these tests were just ported from the Groovy CoreTracer tests in APM java code
+
+    @ParameterizedTest
+    @ValueSource(strings = ["RANDOM", "SEQUENTIAL", "SECURE_RANDOM"])
+    fun `M generate id with strategyName and tIdSize bits`(strategyName: String) {
+        val isTid128 = listOf(false, true)
+        isTid128.forEach { tId128b ->
+            val strategy = IdGenerationStrategy.fromName(strategyName, tId128b)
+            val traceIds = (0..32768).map { strategy.generateTraceId() }
+            val checked = HashSet<DDTraceId>()
+
+            traceIds.forEach { traceId ->
+                assertThat(traceId).isNotNull
+                assertThat(traceId).isNotEqualTo("foo")
+                assertThat(traceId).isNotEqualTo(DDTraceId.ZERO)
+                assertThat(traceId).isEqualTo(traceId)
```

Comment:
/blocking

this is always going to be true.

Reply:
Assuming Sequential here is what we default do. Could you make sure we assert on the right things here

Something like https://github.com/DataDog/dd-sdk-ios/blob/725e59b118935100c36fe90792c3cb2a26438fe2/DatadogInternal/Tests/NetworkInstrumentation/TraceIDGeneratorTests.swift#L22-L36

Reply:
yes I ported this test from the APM code, that's weird indeed let me check what they have in original groovy test. I might have missed something

Reply:
I checked, all good that's the test, I think the reason why is to test the reflexivity of `equals` method: "an object must equal itself"